Reason: actually useful manpages.
Upstream: yes, in master.

The patch below is a heavily trimmed version of it though.
Just bin_search_tree remains.

From 91315195faa54b42e77d4b62b891ec3a0842f16b Mon Sep 17 00:00:00 2001
From: redi <redi@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 10 Dec 2015 14:02:52 +0000
Subject: [PATCH] Improve generated libstdc++ API docs

	* doc/doxygen/user.cfg.in: Use EXTENSION_MAPPING tag. Add new headers
	to INPUT. Remove obsolete XML_SCHEMA and XML_DTD tags. Update
	PREDEFINED macros. Set BRIEF_MEMBER_DESC for man-pages.
	* include/backward/strstream: Correct @file comment.
	* include/bits/forward_list.h: Improve Doxygen comments.
	* include/bits/locale_facets_nonio.h: Likewise.
	* include/bits/mutex.h: Likewise.
	* include/bits/shared_ptr.h: Likewise.
	* include/bits/stl_deque.h: Likewise.
	* include/debug/vector (_Safe_vector): Add @brief section to comment.
	* include/experimental/bits/fs_dir.h: Correct @file comment.
	* include/experimental/bits/fs_fwd.h: Likewise.
	* include/experimental/bits/fs_ops.h: Likewise.
	* include/experimental/bits/fs_path.h: Likewise.
	* include/experimental/bits/string_view.tcc: Likewise.
	* include/experimental/optional: Document experimental status.
	* include/experimental/string_view: Correct @file comment.
	* include/ext/pb_ds/detail/bin_search_tree_/traits.hpp: Reduce
	whitespace to avoid Doxygen bug.
	* include/std/bitset: Remove redundant @class Doxygen command. Add
	parentheses to avoid Doxygen bug.
	* include/std/mutex: Improve Doxygen comments.
	* include/tr2/dynamic_bitset: Add missing @param documentation.
	* scripts/run_doxygen: Rename man pages for std::experimental types.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@231512 138bc75d-0d04-0410-961f-82ee72b054a4
---
 libstdc++-v3/ChangeLog                             | 27 +++++++++++
 libstdc++-v3/doc/doxygen/user.cfg.in               | 54 +++++++++++++++-------
 libstdc++-v3/include/backward/strstream            |  6 +--
 libstdc++-v3/include/bits/forward_list.h           |  5 +-
 libstdc++-v3/include/bits/locale_facets_nonio.h    |  2 +-
 libstdc++-v3/include/bits/mutex.h                  | 23 +++++++--
 libstdc++-v3/include/bits/shared_ptr.h             |  2 +-
 libstdc++-v3/include/bits/stl_deque.h              |  1 +
 libstdc++-v3/include/debug/vector                  |  9 ++--
 libstdc++-v3/include/experimental/bits/fs_dir.h    |  2 +-
 libstdc++-v3/include/experimental/bits/fs_fwd.h    |  7 +--
 libstdc++-v3/include/experimental/bits/fs_ops.h    |  5 +-
 libstdc++-v3/include/experimental/bits/fs_path.h   |  2 +-
 .../include/experimental/bits/string_view.tcc      |  2 +-
 libstdc++-v3/include/experimental/optional         |  6 +++
 libstdc++-v3/include/experimental/string_view      |  2 +-
 .../ext/pb_ds/detail/bin_search_tree_/traits.hpp   |  9 +---
 libstdc++-v3/include/std/bitset                    |  4 +-
 libstdc++-v3/include/std/mutex                     |  4 +-
 libstdc++-v3/include/tr2/dynamic_bitset            |  3 ++
 libstdc++-v3/scripts/run_doxygen                   | 17 +++++++
 21 files changed, 140 insertions(+), 52 deletions(-)

diff --git a/libstdc++-v3/include/ext/pb_ds/detail/bin_search_tree_/traits.hpp b/libstdc++-v3/include/ext/pb_ds/detail/bin_search_tree_/traits.hpp
index da01e1a..674fa92 100644
--- a/libstdc++-v3/include/ext/pb_ds/detail/bin_search_tree_/traits.hpp
+++ b/libstdc++-v3/include/ext/pb_ds/detail/bin_search_tree_/traits.hpp
@@ -166,13 +166,8 @@ namespace __gnu_pbds
 	     class Node_Update,
 	     class Node,
 	     typename _Alloc>
-    struct bin_search_tree_traits<
-      Key,
-      null_type,
-      Cmp_Fn,
-      Node_Update,
-      Node,
-      _Alloc>
+    struct
+    bin_search_tree_traits<Key, null_type, Cmp_Fn, Node_Update, Node, _Alloc>
     {
     private:
       typedef types_traits<Key, null_type, _Alloc, false> type_traits;
